### PR TITLE
Fix PNP loss to make it work with negatives without related positives

### PR DESCRIPTION
**Pull request by Puzer**
Currently PNP loss returns NaN if you have some negatives examples without related positive examples 
`
labels = torch.tensor([1, 1, 2])
`

Let's say you have anchor, positive and negative.
N_pos (from PNP loss) for labels in this case will be [2, 2, 0]
So in this case you will get devision by 0 and loss will be NaN as result.

This fix keeps only positive instances at the final stage.

I've tested that for my use-case and it works quite well.
However I'm not sure wether it's mathematically correct or not, maybe there is more reasonable fix of this issue.

### CODE REVIEW

**Comment by KevinMusgrave** [comment]
@interestingzhuo Any thoughts on this?

**Comment by interestingzhuo** [comment]
> @interestingzhuo Any thoughts on this?
That's right! And the code will be 
```
loss = torch.sum(sim_all_rk * I_pos, dim=-1)[safe_N] / N_pos[safe_N].reshape(-1)
loss = torch.sum(loss) / torch.sum(safe_N)
```
for effective training.

**Comment by KevinMusgrave** [comment]
Thanks @Puzer and @interestingzhuo !
